src/config/overrides.ts

```diff
@@ -56,4 +56,27 @@ export const overrides = [
       ],
     },
   },
+  // We normally disallow process.env, but it's fine to use it in test files,
+  // config files, scripts, etc., and also in the `env.ts` file.
+  {
+    files: [
+      "*.test.ts",
+      "*.test.tsx",
+      "*.type-test.ts",
+      "*.type-test.tsx",
+      "**/test/**",
+      "**/.jest/**",
+      "**/env/**/*.ts",
+      "**/env.ts",
+      "**/environment.ts",
+      "**/next-config/**",
+      "**/webpack/**",
+      "scripts/**",
+      "jest.config.*",
+      "webpack.config.*",
```

Comment:
These are all oddly specific. I’d leave these to be per-project, or at least keep them to our monorepo config, not add them to the generic plugin.